### ISSUE TITLE
JOGL 3D Rendering broken on macOS >= 10.14

### ISSUE THREAD

**Issue by Spiess** [vitrivr/cineast#126]
Feature extractors and exporters using the JOGLOffscreenRenderer crash on macOS >= 10.14 with the error `NSWindow drag regions should only be invalidated on the Main Thread!` as described in [this forum thread](http://forum.jogamp.org/JOGL-on-Mojave-Mac-OS-10-14-td4039037.html).

The issue is [fixed in the 2.4.0](https://jogamp.org/bugzilla//show_bug.cgi?id=1370) version of JOGL, but this version is still in development and not available as release version.

**Comment by lucaro** [comment]
As there appears no clear timeline for the release of this next library version, is there a workaround which could be used in the meantime?

**Comment by Spiess** [comment]
Maybe [jGL](https://github.com/jzy3d/jGL) could be investigated as an alternative.

**Comment by Spiess** [comment]
What about [Three.kt](https://github.com/markaren/three.kt)?
From what I understand it is built for real time rendering, like Three.js, but it should probably also work for static rendering.

**Comment by lucaro** [comment]
Interesting. I just checked, it depends on [lwjgl](https://github.com/LWJGL/lwjgl3) for OpenGL which is better maintained than JOGL, offers access to all kinds of things but is heavier. If anything, we could just depend on lwjgl directly.

**Comment by Spiess** [comment]
You are right, using lwjgl makes a lot more sense. It even has macOS aarch64 build artifacts.
